pci/src/vfio.rs

```diff
@@ -274,6 +275,57 @@ pub struct MmioRegion {
     pub(crate) index: u32,
     pub(crate) user_memory_regions: Vec<UserMemoryRegion>,
 }
+
+trait MmioRegionRange {
+    fn check_range(&self, guest_addr: u64, size: u64) -> bool;
```

Comment:
nit: it might be good to the the `GuestAddress` and `GuestUSize` types in this trait.

Reply:
I was debating whether to do that, it seems implied that GuestAddress would be compatible and initialized with calls into the vm_memory library so I felt that it might be best to keep this separate. Additionally user_memory_region, which this references, utilizes u64 for its addresses.